src/Transform/ONNX/Decompose.cpp

```diff
@@ -397,47 +363,67 @@ Value insertPadAxis(PatternRewriter &rewriter, Location loc, Value input,
   }
   padResults.emplace_back(splitLastResults);
   // Concat padded results.
-  Value concatResult = emitConcat(rewriter, loc, ValueRange(padResults), axis);
+  onnx_mlir::MultiDialectBuilder<onnx_mlir::OnnxBuilder> create(rewriter, loc);
+  Type elementType =
+      padResults[0].getType().cast<ShapedType>().getElementType();
+  Type concatType = UnrankedTensorType::get(elementType);
+  Value concatResult =
+      create.onnx.concat(concatType, ValueRange(padResults), axis);
   return concatResult;
 }
 
 // Insert pads between elements in input tensor in spatial dimensions.
 // The padding size is strides - 1
-Value insertPadsConvTransposeInput(
-    PatternRewriter &rewriter, Location loc, Value input, ArrayAttr strides) {
+Value insertPadsConvTransposeInput(PatternRewriter &rewriter, Location loc,
+    ONNXConvTransposeOp op, Value input) {
+  ONNXConvTransposeOpShapeHelper shapeHelper(op.getOperation(), {});
+  LogicalResult shapecomputed = shapeHelper.computeShape();
+  assert(succeeded(shapecomputed) &&
+         "unexpected inferShapes failuer for ConvTrans op");
+  SmallVector<int64_t, 2> strides = shapeHelper.strides;
   int64_t spatialOffset = 2;
   for (unsigned int i = 0; i < strides.size(); ++i) {
     input = insertPadAxis(rewriter, loc, input, /*axis*/ spatialOffset + i,
-        /*padSize*/ ArrayAttrIntVal(strides, i) - 1);
+        /*padSize*/ strides[i] - 1);
   }
   return input;
 }
 
 // Insert additional padding to output of ConvOp in ConvTransposeOp.
 Value insertAdditionalPadsConvTranspose(PatternRewriter &rewriter, Location loc,
-    ONNXConvOp op, Value input, ArrayAttr outputShapeAttr) {
-  ONNXConvOpShapeHelper shapeHelper(op.getOperation(), {});
-  shapeHelper.computeShapeAndAssertOnFailure();
-  int inputRank = shapeHelper.getOutputDims().size();
+    ONNXConvOp convOp, Value input, ONNXConvTransposeOp op) {
+  ONNXConvOpShapeHelper convShapeHelper(convOp.getOperation(), {});
+  Type elementType = input.getType().cast<ShapedType>().getElementType();
+  assert(succeeded(convShapeHelper.computeShapeAndUpdateType(elementType)) &&
+         "unexpected inferShapes failure for Conv op");
```

Comment:
`computeShapeAndUpdateType` has assertion. There is no need to assert here.

Reply:
Done. Thanks.